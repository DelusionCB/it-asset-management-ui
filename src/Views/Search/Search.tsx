import React, {useState} from 'react';
import {Container, Spinner} from 'reactstrap';
import Searchbar from '../../Components/Searchbar/Searchbar';
import {useTranslation} from 'react-i18next';
import {getEndpointData} from '../../Api/Utils/fetchUtils';
import ItemGrid from '../../Components/ItemGrid/ItemGrid';
import './index.scss'

function Search (): JSX.Element {
    const [results, setResults] = useState([])
    const [completed, setCompleted] = useState(false)
    const [loading, setLoading] = useState(false)
    const {t} = useTranslation()

    async function executeSearch (query: string, type: string, fetch: boolean): Promise<void> {
        if (!fetch) {
            setResults([])
        } else {
            setLoading(true)
            try {
                const response = await getEndpointData({endpoint: type, filter: query})
                setResults(response.results)
                setCompleted(true)
            } finally {
                setLoading(false)
            }
        }
    }

    function getResults (): JSX.Element | null {
        return completed && results.length < 0
            ? null
            : <ItemGrid t={t} items={results} />
    }

    return (
        <Container className='search-page'>
            <Searchbar
                onFormSubmit={(query, type, fetch) => { void executeSearch(query, type, fetch) }}
                t={t}
            />
            <div>
                <div className="results-count-status" role="status" aria-live="polite" aria-atomic="true">
                    {completed &&
                        <p>{t('search-results', {count: results.length})}</p>
                    }
                </div>
                <section className="container-fluid">
                    {loading
                        ? <div className="search-loading-spinner">
                            <Spinner animation="border" role="status">
                                Ladataan...
                            </Spinner>
                        </div>
                        : getResults()
                    }
                </section>
            </div>
        </Container>
    );
}

export default Search;
