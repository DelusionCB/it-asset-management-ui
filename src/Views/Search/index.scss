.search-page {
    .results-count-status {
        margin: 1rem 0 1rem 0;
        padding-right: calc(1.5rem * .5);
        padding-left: calc(1.5rem * .5);
    }
    .search-bar {
        button {
            background-color: var(--primary-color);
            border-radius: 0;
            font-size: 1.25rem;
            border: 2px solid black;
            &:hover, &:focus {
                border: 2px solid var(--secondary-color);
                background-color: var(--primary-color);
            }
        }
    }
    .selection-header {
        h1 {
            font-size: 1.5rem !important;
        }
    }
    .search-header {
        margin-bottom: 1.25rem;
    }
}
