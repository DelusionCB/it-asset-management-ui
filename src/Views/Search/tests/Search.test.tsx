import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import Search from '../Search';
import {BrowserRouter} from 'react-router-dom'
import '@testing-library/jest-dom'

describe('<Search />', () => {
    let renderer: TestRenderer.ReactTestRenderer;

    // Render the component before tests
    beforeAll(() => {
        renderer = TestRenderer.create(
            <BrowserRouter>
                <Search />
            </BrowserRouter>,
        );
    });
    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
