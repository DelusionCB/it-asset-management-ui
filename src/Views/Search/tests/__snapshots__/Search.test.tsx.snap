// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> Snapshot renders correctly 1`] = `
<div
  className="search-page container"
>
  <div
    className="container"
  >
    <h1>
      search-header
    </h1>
    <div
      className="row"
    >
      <div
        className="selection-wrapper"
      >
        <div
          className="selection-header"
        >
          <h1>
            select-archive-search
          </h1>
        </div>
        <div
          className="selection-radios"
        >
          <div>
            <div
              className="form-check"
            >
              <label
                className="form-check-label"
                htmlFor="application-0"
              >
                radios.application
              </label>
              <input
                aria-disabled={false}
                checked={true}
                className="form-check-input"
                disabled={false}
                id="application-0"
                name="application"
                onChange={[Function]}
                type="radio"
              />
            </div>
          </div>
          <div>
            <div
              className="form-check"
            >
              <label
                className="form-check-label"
                htmlFor="directory-1"
              >
                radios.directory
              </label>
              <input
                aria-disabled={false}
                checked={false}
                className="form-check-input"
                disabled={false}
                id="directory-1"
                name="directory"
                onChange={[Function]}
                type="radio"
              />
            </div>
          </div>
          <div>
            <div
              className="form-check"
            >
              <label
                className="form-check-label"
                htmlFor="license-2"
              >
                radios.license
              </label>
              <input
                aria-disabled={false}
                checked={false}
                className="form-check-input"
                disabled={false}
                id="license-2"
                name="license"
                onChange={[Function]}
                type="radio"
              />
            </div>
          </div>
          <div>
            <div
              className="form-check"
            >
              <label
                className="form-check-label"
                htmlFor="server-3"
              >
                radios.server
              </label>
              <input
                aria-disabled={false}
                checked={false}
                className="form-check-input"
                disabled={false}
                id="server-3"
                name="server"
                onChange={[Function]}
                type="radio"
              />
            </div>
          </div>
          <div>
            <div
              className="form-check"
            >
              <label
                className="form-check-label"
                htmlFor="service-4"
              >
                radios.service
              </label>
              <input
                aria-disabled={false}
                checked={false}
                className="form-check-input"
                disabled={false}
                id="service-4"
                name="service"
                onChange={[Function]}
                type="radio"
              />
            </div>
          </div>
          <div>
            <div
              className="form-check"
            >
              <label
                className="form-check-label"
                htmlFor="provider-5"
              >
                radios.provider
              </label>
              <input
                aria-disabled={false}
                checked={false}
                className="form-check-input"
                disabled={false}
                id="provider-5"
                name="provider"
                onChange={[Function]}
                type="radio"
              />
            </div>
          </div>
          <div>
            <div
              className="form-check"
            >
              <label
                className="form-check-label"
                htmlFor="contract-6"
              >
                radios.contract
              </label>
              <input
                aria-disabled={false}
                checked={false}
                className="form-check-input"
                disabled={false}
                id="contract-6"
                name="contract"
                onChange={[Function]}
                type="radio"
              />
            </div>
          </div>
          <div>
            <div
              className="form-check"
            >
              <label
                className="form-check-label"
                htmlFor="integration-7"
              >
                radios.integration
              </label>
              <input
                aria-disabled={false}
                checked={false}
                className="form-check-input"
                disabled={false}
                id="integration-7"
                name="integration"
                onChange={[Function]}
                type="radio"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="search-bar"
      >
        <form
          className=""
          onSubmit={[Function]}
        >
          <div
            className="textfield col"
          >
            <div
              className="mb-3"
            >
              <label
                className="form-label"
                htmlFor="search"
              >
                archive-contents
              </label>
              <input
                aria-invalid={false}
                className="form-control"
                disabled={false}
                id="search"
                name="search"
                onChange={[Function]}
                placeholder="archive-example"
                type="text"
                value=""
              />
            </div>
          </div>
        </form>
        <button
          className="btn btn-secondary disabled"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          <span>
            archive-fetch
          </span>
        </button>
      </div>
    </div>
  </div>
  <div>
    <div
      aria-atomic="true"
      aria-live="polite"
      className="results-count-status"
      role="status"
    />
    <section
      className="container-fluid"
    >
      <div
        className="row event-grid"
      />
    </section>
  </div>
</div>
`;
